automation/src/test/scala/org/broadinstitute/dsde/test/api/BillingProjectOwnerPermissionSpec.scala

```diff
@@ -0,0 +1,270 @@
+import org.broadinstitute.dsde.workbench.service.Orchestration.groups.GroupRole
+import org.broadinstitute.dsde.workbench.service.{Rawls, RestException, Sam}
+import org.scalatest.{FreeSpec, Matchers}
+
+import scala.util.Try
+
+class BillingProjectOwnerPermissionSpec extends FreeSpec with WorkspaceFixtures with BillingFixtures with GroupFixtures  with Matchers {
+
+  /*
+ * Unless otherwise declared, this auth token will be used for API calls.
+ * We are using a curator to prevent collisions with users in tests (who are Students and AuthDomainUsers), not
+ *  because we specifically need a curator.
+ */
+
+  val defaultUser: Credentials = UserPool.chooseCurator
+  val authTokenDefault: AuthToken = defaultUser.makeAuthToken()
+
+
+  "removing permissions from billing project owners for workspaces with auth domains" - {
+
+    "+ project owner, + group member, create workspace, - group member" in {
+      val owner = UserPool.chooseProjectOwner
+      val creator = UserPool.chooseStudent
+      val user = owner
+      implicit val token: AuthToken = creator.makeAuthToken()
+
+      withCleanBillingProject(owner, List(creator.email)) { projectName =>
+        withGroup("group", List(user.email)) { groupName =>
+          withWorkspace(projectName, "GroupApiSpec_workspace", Set(groupName)) { workspaceName =>
+            AuthDomainMatcher.checkVisibleAndAccessible(projectName, workspaceName, List(groupName))(user.makeAuthToken())
+            Sam.user.removeUserFromPolicy(groupName, GroupRole.Member.toString, user.email)
+            AuthDomainMatcher.checkVisibleNotAccessible(projectName, workspaceName)(user.makeAuthToken())
+          }
+        }
+      }
+    }
+
+    "+ project owner, + group member, create workspace, - project owner" in {
+      val owner = UserPool.chooseProjectOwner
+      val creator = owner
+      val user = UserPool.chooseStudent
+      implicit val token: AuthToken = creator.makeAuthToken()
+
+      withCleanBillingProject(creator, ownerEmails = List(user.email)) { projectName =>
+        withGroup("group", List(user.email)) { groupName =>
+          withWorkspace(projectName, "GroupApiSpec_workspace", Set(groupName)) { workspaceName =>
+            AuthDomainMatcher.checkVisibleAndAccessible(projectName, workspaceName, List(groupName))(user.makeAuthToken())
+
+            Rawls.billing.removeUserFromBillingProject(projectName, user.email, BillingProjectRole.Owner)
+            AuthDomainMatcher.checkNotVisibleNotAccessible(projectName, workspaceName)(user.makeAuthToken())
+          }
+        }
+      }
+    }
+
+    "+ project owner, create workspace, + group member, - group member" in {
+      val owner = UserPool.chooseProjectOwner
+      val creator = UserPool.chooseStudent
+      val user = owner
+      implicit val token: AuthToken = creator.makeAuthToken()
+
+      withCleanBillingProject(owner, List(creator.email)) { projectName =>
```

Comment:
I'm seeing inconsistency around naming `owner` and `creator` and `user` between these tests.

Reply:
a student can be either a `owner` or `creator` of a billing project or is both `owner` and `creator` of a billing project. I think naming is fine.

Reply:
It is not consistent with other tests in this class - for example the test right above it defines the 'creator' as the project owner and the 'user' as the student. _This_ test has the 'owner' as the project owner and the 'creator' as a student. The naming should be consistent from test to test. Either use `owner` and `creator` for all the tests or use `owner` and `user` for all the tests, but not a mixture of both

Reply:
👍 use `owner` and `creator` for all tests

Reply:
I still find the `owner` vs `creator` naming confusing. Seeing 
```
val creator = UserPool.chooseProjectOwner
val owner = UserPool.chooseStudent
```
and then
```
val owner = UserPool.chooseProjectOwner
val creator = UserPool.chooseStudent
```
is hard to follow. A more descriptive name might help me because it is not apparent what the `owner` actually owns and what the `creator` has actually created.

Reply:
I wasn't the original author of tests so I'm guessing names meaning are:
creator - new owner of a clean billing project and also is project owner. this user creates new workspace and groups
owner - added as owner of billing project after creator claimed it. in some tests, added as a member of group